aim/agent/aid/universes/k8s/k8s_watcher.py

```diff
@@ -133,50 +135,98 @@ def _renew_klient_watch(self):
+        for ts in self._observe_thread_state.values():
+            ts['watch_stop'] = True
+        if self.klient.watch:
+            self.klient.stop_watch()
+        if exc:
+            raise exc
+
+    def _start_observers(self):
         self._reset_trees()
         self._renew_klient_watch()
-        for event in self.klient.watch.stream(self.wrap_list_namespaced_aci,
-                                              namespace=self.namespace):
-            event = self._parse_event(event)
-            LOG.debug("Kubernetes event received: %s", event)
-            if event:
-                self.affected_tenants |= set(self._process_event(event))
-            self.q.put(object())
+
+        self._observe_thread_state = {}
+        for typ in self._k8s_types_to_observe:
+            thd = threading.Thread(target=self._observe_objects,
+                                   args=(typ,))
+            thd.daemon = True
+            self._observe_thread_state[thd] = dict(watch_stop=False)
+            thd.start()
+
+    def _check_observers(self):
+        exc = None
+        for t in self._observe_thread_state:
+            tstate = self._observe_thread_state[t]
+            if tstate.get('watch_exception'):
+                exc = tstate.get('watch_exception')
+                LOG.info('Thread %s raised exception %s', t, exc)
+                break
+            if tstate.get('http_resp') and tstate['http_resp'].closed:
+                LOG.info('HTTP response closed for thread %s', t)
+                exc = K8SObserverStopped()
+                break
+            if not t.is_alive():
+                LOG.info('Thread %s is not alive', t)
+                exc = K8SObserverStopped()
+                break
+        return exc
+
+    def wrap_list_call(self, *args, **kwargs):
+        tstate = kwargs.pop('_thread_state')
+        resp = self.klient.list(*args, **kwargs)
+        if hasattr(resp, 'close') and callable(resp.close):
+            tstate['http_resp'] = resp
+        return resp
+
+    def _observe_objects(self, k8s_type):
+        my_thread = threading.current_thread()
+        my_state = self._observe_thread_state[my_thread]
+        LOG.debug('Start observing %s objects (thread %s)',
+                  k8s_type.kind, my_thread)
+        if not my_state['watch_stop']:
+            try:
+                kwargs = {}
+                if k8s_type == api_v1.Namespace:
+                    kwargs['name'] = self.namespace
+                for event in self.klient.watch.stream(
+                        self.wrap_list_call, k8s_type,
+                        _thread_state=my_state,
+                        namespace=self.namespace):
+                    if my_state['watch_stop']:
+                        LOG.debug('Stopping %s objects thread %s',
+                                  k8s_type.kind, my_thread)
+                        break
+                    LOG.debug("Kubernetes event (%s objects) received: %s",
```

Comment:
Is it worth having an info level log with some basic info of the event received? Like kind, name and status?